dissect/target/plugins/apps/browser/firefox.py

```diff
@@ -8,16 +13,31 @@
 
 from dissect.target.exceptions import FileNotFoundError, UnsupportedPluginError
 from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
 from dissect.target.helpers.record import create_extended_descriptor
 from dissect.target.plugin import export
 from dissect.target.plugins.apps.browser.browser import (
     GENERIC_COOKIE_FIELDS,
     GENERIC_DOWNLOAD_RECORD_FIELDS,
     GENERIC_HISTORY_RECORD_FIELDS,
+    GENERIC_PASSWORD_RECORD_FIELDS,
     BrowserPlugin,
     try_idna,
 )
 
+try:
+    from asn1crypto import algos, core
```

Comment:
It's a little bit of bloat, but can you separate this into a separate `try/except/HAS_ASN1` block and put error messages appropriately in the functions?